### DISCUSSION TITLE
Override cache-control headers for static pages

### DISCUSSION THREAD

**Discussion by robhrt7** [vercel/next.js#27746] [Help] [answered]
When generating static pages with `getStaticProps`, Next.js automatically uses the following cache headers in response for these static page requests `s-maxage=60, stale-while-revalidate`, where `s-maxage` seems to be equals to `revalidate` config. The problem is that `stale-while-revalidate` is not working, without the value set in seconds, with different reverse-proxy CDN's like Fastly and Cloudflare.

Is there a way to specifically override the cache headers set for static pages to set a custom value for the `stale-while-revalidate`?

As I understand, it's impossible to override cache headers as docs suggest:

> Cache-Control headers set in next.config.js will be overwritten in production to ensure that static assets can be cached effectively. If you need to revalidate the cache of a page that has been statically generated, you can do so by setting revalidate in the page's getStaticProps function.

**Comment by leerob** [comment, marked as answer]
It's now possible to customize `cache-control` as of v14.2.10. Please upgrade and give it a try!

https://github.com/vercel/next.js/pull/69802
